CategoryDto and Mapper

ilya-frolov opened the pull request:
1. Create CategoryDto.
2. Add ModelMapper dependency to pom.xml.
3. Create MyModelMapper.
4. Input into SavedGameDto CategoryDtos instead of Categories.
5. Change saveGame(), getAllGames(), getGameById(), updateGame() in GameServiceImpl: provide CategoryDtos to SavedGameDto.fromGame() by modelMapper.toCategoryDto() method.
6. Change all @Autowired fields to final and add @RequiredArgsConstructor to GameServiceImpl.

ilya-frolov committed:
1. Create CategoryDto.
2. Add ModelMapper dependency to pom.xml.
3. Create MyModelMapper.
4. Input into SavedGameDto CategoryDtos instead of Categories.
5. Change saveGame(), getAllGames(), getGameById(), updateGame() in GameServiceImpl: provide CategoryDtos to SavedGameDto.fromGame() by modelMapper.toCategoryDto() method.
6. Change all @Autowired fields to final and add @RequiredArgsConstructor to GameServiceImpl.

RomanKashenok commented on src/main/java/info/freeit/boarderpicker/mapper/MyModelMapper.java:
```diff
@@ -0,0 +1,20 @@
+public class MyModelMapper {
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    public Set<CategoryDto> toCategoryDto(Set<Category> categories) {
```
если пользуешься маппером - мапь всё тут. А-то у тебя игра где-то в одном месте мапится, в ней категории в маппере - как-то не складывается

ilya-frolov replied:
Исправил.

ilya-frolov committed:
1. Create SavedGameDto by MyModelMapper instead of fromGame() method in SavedGameDto.
2. Delete fromGame() from SavedGameDto.
3. Change saveGame(), getAllGames(), getGameById(), updateGame() in GameServiceImpl: modelMapper.for creating SavedGameDto.

RomanKashenok reviewed (approved):
Апрув, но надо поменять на нормальный модел маппер (что б работало пока хватит)
